testcafe/tests/drive/folder_sharing_scenario.js

```diff
@@ -0,0 +1,130 @@
+import { Selector, Role } from 'testcafe'
+import { driveUser } from '../helpers/roles'
+import {
+  TESTCAFE_DRIVE_URL,
+  isExistingAndVisibile,
+  getCurrentDateTime,
+  FOLDER_DATE_TIME
+} from '../helpers/utils'
+import Data from '../helpers/data'
+const data = new Data()
+
+import DrivePage from '../pages/drive-model'
+const drivePage = new DrivePage()
+
+import PublicDrivePage from '../pages/drive-model-public'
+const publicDrivePage = new PublicDrivePage()
+
+const file = '[cozy]QA_table_ APPS.pdf'
+
+//************************
+//Tests when authentified
+//************************
+fixture`Folder link Sharing Scenario`.page`${TESTCAFE_DRIVE_URL}/`.beforeEach(
+  async t => {
+    await t.useRole(driveUser)
+    await drivePage.waitForLoading()
+  }
+)
+
+test('Drive : Create a $test_date_time folder in Drive', async t => {
+  await drivePage.addNewFolder(FOLDER_DATE_TIME)
+  //We need to pass FOLDER_DATE_TIME through multiple fixture, so we cannot use ctx here.
+})
+
+test('Drive : from Drive, go in a folder, upload a file, and share the folder', async t => {
+  await drivePage.goToFolder(FOLDER_DATE_TIME)
+  await drivePage.openActionMenu()
+  await t.pressKey('esc') //close action Menu
+
+  await drivePage.uploadFiles([`../../data/${file}`])
+  await drivePage.shareFolderPublicLink()
+
+  const link = await drivePage.copyBtnShareByLink.getAttribute('data-test-url')
+  if (link) {
+    data.sharingLink = link
+    console.log(`SHARING_LINK : ` + data.sharingLink)
+  }
+})
+
+//************************
+// Public (no authentification)
+//************************
+fixture`Drive : Access a folder public link`.page`${TESTCAFE_DRIVE_URL}/`
+  .beforeEach(async t => {
+    await t.useRole(Role.anonymous())
+  })
+  .afterEach(async t => {
+    await publicDrivePage.checkLocalFile('files.zip')
+    await publicDrivePage.deleteLocalFile('files.zip')
+  })
+test('Drive : Access a folder public link (desktop)', async t => {
```

Comment:
idem it's more about downloading than accessing, no? We don't check if something specific to this file is displayed on the screen, right?

Reply:
We don't check something specific to the file, just that there are buttons to download it, and to create a new cozy.

Reply:
So I would prefer to have: 

test('Download a folder from a public link') 

test('Create a cozy from a public link')

Reply:
new name : *Drive : Access a folder public link, download the file(s), and check the 'create Cozy' link`*